### PR TITLE
fix(pdf): keep ordered list markers clear of body text

### PR DESCRIPTION
**Pull request by amruthpillai**
Ordered list markers could overlap their body text once labels reached two or three digits; even single-digit Courier markers exceeded the fixed gutter. Ordered markers now use intrinsic text width plus a common minimum gutter for each list, accounting for marker font size and letter spacing so body starts stay aligned across 9→10 and 99→100. Existing list pagination tags and behavior are preserved.

This addresses overlap reproduced during the investigation of #2751. The original v5.0.10 missing-leading-digit screenshot was not reproduced, and this PR does not claim to resolve that exact historical symptom. Authored ordered-list start remains unchanged. Existing RTL nested-list flattening also remains unchanged (baseline/fixed PDF coordinates match).

Validation: 30 actual-PDF regression tests cover Helvetica, Courier and Noto Serif SC; one-, two- and three-digit labels; LTR/RTL; narrow columns/sidebar; LTR nesting; page breaks and content retention; authored font size/inheritance, letter spacing and row gaps. All 36 existing list pagination tests pass. Independent review reproduced the letter-spacing edge case and verified the correction.

Full PDF suite: 932 tests passed. PDF typecheck, package boundaries, and repository checks passed.

Refs #2751

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved ordered-list formatting in generated PDFs by maintaining consistent spacing between list markers and body text.
  * Marker spacing now adapts across different font sizes, letter spacing, list lengths, layouts, and text directions.

<details open><summary><h3>Greptile Summary</h3></summary>

The PR changes ordered-list marker sizing in generated PDFs so every item reserves a common minimum gutter based on list length, marker font size, and letter spacing.
- Preserves explicit marker widths and existing list pagination behavior.
- Adds PDF regression coverage across multiple fonts, list lengths, directions, layouts, nesting, pagination, and authored typography.
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

The PR appears safe to merge.

No blocking failure remains.
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/pdf/src/templates/shared/rich-text.tsx | Computes and caches each ordered list’s marker width floor while preserving authored width and flex-basis overrides. |
| packages/pdf/src/templates/shared/ordered-marker.test.tsx | Adds broad rendered-PDF assertions for marker clearance, aligned body edges, pagination, RTL/LTR output, typography, and row gaps. |

</details>

<sub>Reviews (2): Last reviewed commit: ["perf(pdf): cache ordered list marker siz..."](https://github.com/amruthpillai/reactive-resume/commit/7ac7fd31bd71f180e1d6592fae18c4e541edb102) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=60788764)</sub>

### CODE REVIEW

File: packages/pdf/src/templates/shared/rich-text.tsx [hunk 1]
```diff
@@ -245,6 +245,27 @@ export const RichText = ({ children, semanticField }: RichTextProps) => {
...
+						element.parentNode?.childNodes.filter((child) => child.rawTagName?.toLowerCase() === "li").length ?? 1;
+					const markerFontSize =
+						typeof markerResolved.style?.fontSize === "number"
+							? markerResolved.style.fontSize
+							: metadata.typography.body.fontSize;
+					const markerLetterSpacing =
+						typeof markerResolved.style?.letterSpacing === "number"
+							? Math.max(0, markerResolved.style.letterSpacing)
+							: 0;
+					const markerDigits = String(listLength).length;
```
[the remark this answers is not in the document]
Comment (amruthpillai) [reply packages/pdf/src/templates/shared/rich-text.tsx]: Fixed. Marker length/font/spacing work now runs only for ordered lists without an explicit resolved override, and a per-render WeakMap caches each parent list length so sibling traversal occurs once per list instead of once per item. After rebasing onto merged #3448, all 975 PDF tests and PDF typecheck pass.
